eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/FixImagesVarDirCommand.php

```diff
@@ -0,0 +1,416 @@
+                    $this->moveFile($filePath, $newFilePath);
+                }
+            }
+
+            ++$this->done;
+        }
+
+        if (!$this->dryRun) {
+            $this->updateContentObjectAtributes();
+        }
+    }
+
+    /**
+     * @todo Implement this(?)
+     *
+     * @param string $oldFilePath
+     * @param string $newFilePath
+     */
+    protected function moveFile(string $oldFilePath, string $newFilePath)
+    {
+    }
+
+    /**
+     * @param int $imageId
+     * @param int $contentObjectAttributeId
+     * @param string $oldFilePath
+     * @param string $newFilePath
+     */
+    protected function updateImage(int $imageId, int $contentObjectAttributeId, string $oldFilePath, string $newFilePath)
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query
+            ->update('ezimagefile', 'i')
+            ->set('i.filepath', $query->expr()->literal($newFilePath))
+            ->where('i.id = :id')
+            ->setParameter('id', $imageId);
+
+        $query->execute();
+
+        $this->imageAttributes[$contentObjectAttributeId][$oldFilePath] = $newFilePath;
+    }
+
+    protected function updateContentObjectAtributes()
+    {
+        foreach ($this->imageAttributes as $attributeId => $files) {
+            $attributeObjects = $this->getContentObjectAtrributesById($attributeId);
+
+            foreach ($attributeObjects as $attributeObject) {
+                $dom = new DOMDocument('1.0', 'utf-8');
+
+                try {
+                    $dom->loadXML($attributeObject['data_text']);
+                } catch (\RuntimeException $e) {
+                    continue;
+                }
+
+                foreach ($dom->getElementsByTagName('ezimage') as $ezimageNode) {
+                    $oldPath = $ezimageNode->getAttribute('url');
+
+                    if (isset($files[$oldPath])) {
+                        $ezimageNode->setAttribute('url', $files[$oldPath]);
+                        $ezimageNode->setAttribute('dirpath', \dirname($files[$oldPath]));
+                    }
+
+                    foreach ($ezimageNode->getElementsByTagName('alias') as $ezimageAlias) {
+                        $oldPath = $ezimageAlias->getAttribute('url');
+                        if (isset($files[$oldPath])) {
+                            $ezimageAlias->setAttribute('url', $files[$oldPath]);
+                            $ezimageAlias->setAttribute('dirpath', \dirname($files[$oldPath]));
+                        }
+                    }
+                }
+
+                $this->updateContentObjectAtribute($attributeObject['id'], $attributeObject['version'], $dom->saveXML());
+            }
+        }
+    }
+
+    /**
+     * @param int $id
+     * @param int $version
+     * @param string $dataText
+     */
+    protected function updateContentObjectAtribute(int $id, int $version, string $dataText)
+    {
+        $query = $this->connection->createQueryBuilder();
```

Comment:
Please move all queries to a dedicated gateway. Cleaner solution would require updating persistence handler and setting up some integration testing.